Admin/dataobjects/AdminUser.php

```diff
@@ -180,6 +180,9 @@ public function isAuthenticated(AdminApplication $app)
 					$authenticated = true;
 				}
 			}
+
+			// Make sure instance is set so activation check works properly.
+			$this->setInstance($app->getInstance());
```

Comment:
in a multi-instance admin, when logged in without an instance (e.g. https://hamlin/course-host/work-nick/www/admin/) I get the following error:
```
Catchable fatal error: Argument 1 passed to AdminUser::setInstance() must be an instance of SiteInstance, null given, called in /so/packages/admin/work-nick/Admin/dataobjects/AdminUser.php on line 185 and defined in /so/packages/admin/work-nick/Admin/dataobjects/AdminUser.php on line 355
```

You'll need to check of the result of `$app->getInstance()` is an `instanceof SiteInstance` or use the logic up on line 180 and put your `$this->setInstance($app->getInstance());` there.